### PR TITLE
Support Plugin Framework v1.1.1

### PR DESCRIPTION
**Pull request by t0yv0**
This PR does a quick upgrade of the Plugin Framework support to v1.1.1. It is just enough to pass the previous test suite. Still would be good to read the [changelog](https://github.com/hashicorp/terraform-plugin-framework/blob/main/CHANGELOG.md#111-january-13-2022) to see if there's anything new the bridge is missing altogether, possibly CustomTypes support.

- [x] fill in the stubs in pfutils
- [x] rewrite plan modifiers used in test
- [x] test code does not compile
- [x] no diff on `make build.testproviders`
- [x] make test
- [x] make test.integration

### CODE REVIEW

File: pkg/tfpfbridge/tests/internal/cmd/pulumi-resource-tls/schema.json [hunk 1]
```diff
@@ -74,15 +74,7 @@
...
-            "type": "object",
-            "required": [
-                "fromEnv"
-            ],
-            "language": {
-                "nodejs": {
-                    "requiredInputs": []
-                }
-            }
+            "type": "object"
```
Comment (t0yv0) [inline pkg/tfpfbridge/tests/internal/cmd/pulumi-resource-tls/schema.json:77]: Looks like fromEnv stopped being recognized as required, need to double check why.

Comment (t0yv0) [reply pkg/tfpfbridge/tests/internal/cmd/pulumi-resource-tls/schema.json:77]: OK this is now simply marked `Optional: true` in the upstream version of the tls provider and has nothing to do with the framework changes being made.

**Comment by t0yv0** [comment]
Friendly ping @jazzyfresh would it be possible to review?

**Review by jazzyfresh — APPROVED** [review]
LGTM

Thank you for your patience, it took me a bit of time to read through each commit and understand

**Comment by t0yv0** [comment]
> Thank you for your patience, it took me a bit of time to read through each commit and understand

NP at all, please take your time. Quality over speed here. I'm trying to itemize change into smaller chunks going forward so it's less to digest.

**Pull request merged**